src/main/java/com/melon/portfoliomanager/repositories/UserRepository.java

```diff
@@ -0,0 +1,16 @@
+package com.melon.portfoliomanager.repositories;
+
+import com.melon.portfoliomanager.models.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public interface UserRepository extends JpaRepository<User, Long> {
+
+    @Query("SELECT u FROM User u WHERE u.username = ?1")
```

Comment:
Why do you need to write custom sql ? You can just specify what you want to get from the sql table in the name of the repository method like this:
```List<User> findByUsername(String username);``` and it should work.